src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -291,15 +320,133 @@ public Visitable visit(ExplicitConstructorInvocationStmt node, Void arg) {
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  @SuppressWarnings("nullness:override")
+  public @Nullable Visitable visit(IfStmt n, Void arg) {
+    HashSet<String> localVarInCon = new HashSet<>();
+    localVariables.push(localVarInCon);
+    Expression condition = (Expression) n.getCondition().accept(this, arg);
+    localVariables.pop();
+    localVarInCon = new HashSet<>();
+    localVariables.push(localVarInCon);
+    Statement elseStmt = n.getElseStmt().map(s -> (Statement) s.accept(this, arg)).orElse(null);
+    localVariables.pop();
+    localVarInCon = new HashSet<>();
+    localVariables.push(localVarInCon);
+    Statement thenStmt = (Statement) n.getThenStmt().accept(this, arg);
+    localVariables.pop();
+    if (condition == null || thenStmt == null) {
+      return null;
+    }
+    n.setCondition(condition);
+    @SuppressWarnings(
+        "nullness") // elseStmt could actually be null, according to the document of setElseStmt.
+    // This part is to avoid the false warnings of Checker Framework
+    @NonNull Statement nullableElseStme = elseStmt;
+    n.setElseStmt(nullableElseStme);
+    n.setThenStmt(thenStmt);
+    return n;
+  }
+
+  @Override
+  public Visitable visit(WhileStmt node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(SwitchExpr node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(SwitchEntry node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(TryStmt node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(CatchClause node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(BlockStmt node, Void p) {
+    HashSet<String> currentLocalVariables = new HashSet<>();
+    localVariables.push(currentLocalVariables);
+    super.visit(node, p);
+    localVariables.pop();
+    return node;
+  }
+
+  @Override
+  public Visitable visit(VariableDeclarator decl, Void p) {
+    boolean isAField = false;
+    if (!decl.getParentNode().isEmpty()) {
+      if (decl.getParentNode().get() instanceof FieldDeclaration) {
+        isAField = true;
```

Comment:
Rather than two `if` statements, just set `isAField`'s value to the conjunction of the two conditions.